packages/puppeteer-core/src/common/util.ts

```diff
@@ -708,3 +710,16 @@ export class Mutex {
     resolve();
   }
 }
+
+/**
+ * @internal
+ */
+export function timeout(ms: number): Observable<never> {
+  return ms === 0
```

Comment:
How about `Infinity` rather than `0`?

Reply:
That's incorrect as the Common Puppeteer representation of `No Timeout` is `timeout=0`. 
Also if you use `SetTimeout(()=>{}, Infinity)` Node may complain.
```
TimeoutOverflowWarning: Infinity does not fit into a 32-bit signed integer.
```